DataFormats/L1TParticleFlow/interface/layer1_objs.h

```diff
@@ -59,6 +75,31 @@ namespace l1ct {
       unpack_from_bits(src, start, ret.hwPhi);
       unpack_from_bits(src, start, ret.hwEmPt);
       unpack_from_bits(src, start, ret.hwEmID);
+      unpack_from_bits(src, start, ret.hwSrrTot);
+      unpack_from_bits(src, start, ret.hwMeanZ);
+      unpack_from_bits(src, start, ret.hwHoe);
+      return ret;
+    }
+
+    inline ap_uint<BITWIDTH_SLIM> pack_slim() const {
+      ap_uint<BITWIDTH_SLIM> ret;
+      unsigned int start = 0;
+      pack_into_bits(ret, start, hwPt);
+      pack_into_bits(ret, start, hwEta);
+      pack_into_bits(ret, start, hwPhi);
+      pack_into_bits(ret, start, hwEmPt);
+      pack_into_bits(ret, start, hwEmID);
+      return ret;
+    }
+    inline static HadCaloObj unpack_slim(const ap_uint<BITWIDTH_SLIM> &src) {
+      HadCaloObj ret;
+      ret.clear();
+      unsigned int start = 0;
+      unpack_from_bits(src, start, ret.hwPt);
+      unpack_from_bits(src, start, ret.hwEta);
+      unpack_from_bits(src, start, ret.hwPhi);
+      unpack_from_bits(src, start, ret.hwEmPt);
+      unpack_from_bits(src, start, ret.hwEmID);
```

Comment:
there's a lot of ugly duplication here that can be removed:
* define BITWIDTH_SLIM, and then define BITWIDTH = BITWIDTH_SLIM + size of extra members
* unpacking doesn't need to be duplicated since `ap_uint<BITWIDTH_SLIM>` can be promoted to `ap_uint<BITWIDTH>` automatically
* `pack_slim()` could just `{ return pack()(BITWIDTH_SLIM-1,0); }`